react/week3/todo-week3/src/components/ToDoItemRow.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { Component } from "react";
+import "../App.css";
+import PropTypes from "prop-types";
+
+export class ToDoItemRow extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todoItem: this.props.toDo,
+      isEditing: false,
+      hideUpdateButton: true,
+      hideEditButton: false,
+    };
+
+    this.editItemClicked = this.editButtonClicked.bind(this);
+    this.todoUpdated = this.todoUpdated.bind(this);
+    this.updateButtonClicked = this.updateButtonClicked.bind(this);
+  }
+
+  editButtonClicked(e) {
+    e.preventDefault();
+    this.setState({
+      isEditing: true,
+      hideUpdateButton: false,
+      hideEditButton: true,
+    });
+  }
+
+  updateButtonClicked(e) {
+    e.preventDefault();
+    this.setState({
+      isEditing: false,
+      hideUpdateButton: true,
+      hideEditButton: false,
+    });
+  }
+
+  todoUpdated(e) {
```

Comment:
would this make sense?

```suggestion
  updateTodoTiem(e) {
```